framework/material/FileSelectorDialog.kt

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright (C) 2022 Vaticle
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ */
+
+package com.vaticle.typedb.studio.framework.material
+
+import androidx.compose.ui.awt.ComposeDialog
+import com.vaticle.typedb.studio.state.common.util.Property
+import java.awt.FileDialog
+import java.io.File
+import java.util.Optional
+import javax.swing.JFileChooser
+
+object FileSelectorDialog {
+    fun selectFilePath(parent: ComposeDialog, title: String, selectorOptions: SelectorOptions): String? {
+        val file = when (Property.OS.Current) {
+            Property.OS.MACOS -> macOSFileSelector(parent, title, selectorOptions)
+            else -> otherOSFileSelector(title, selectorOptions)
+        }
+
+        return if (file.isEmpty) null else file.get().absolutePath
+    }
+
+    private fun macOSFileSelector(parent: ComposeDialog, title: String, selectorOptions: SelectorOptions): Optional<File> {
+        when (selectorOptions) {
+            SelectorOptions.FILES_ONLY -> System.setProperty("apple.awt.fileDialogForDirectories", "false")
+            SelectorOptions.DIRECTORIES_ONLY -> System.setProperty("apple.awt.fileDialogForDirectories", "true")
```

Comment:
Is there any place in the application that we open any kind of file dialog, besides `FileSelectorDialog.selectFilePath`?

If there is, then this could potentially "silently" affect the behaviour of those dialogs.

Reply:
The only other location we use `FileDialog` is for saving files. Our change doesn't affect this behaviour, and this has been tested on all platforms - we use the `FileDialog` for all platforms when saving rather than splitting between making use of the `JFileChooser` and this behaviour change on macOS does not change where you can save.

Reply:
Should we expand the scope of `FileSelectorDialog` to include saving, perhaps to `FileDialog`? If so, we should look into why this design decision was made in the first place.

Reply:
There's a good reason we've structured it this way: we want restrictions on loading certain files, but we don't need any restrictions on saving files. `FileDialog` looks better, so we use it for saving where we don't care for the restrictions and only use `JFileChooser` where we need restrictions on being able to load files xor directories. As such, we'll keep it as is.